File: 백준/그리디/1946.py
import sys
def solution(arr):
   ans = []
   for i in arr:
      i.sort(key = lambda x : (x[0]))
      end = i[0][1]
      count = 1
      for s,e in i:
         if end>e:
            count +=1
            end = e
      ans.append(count)
   return ans


n = int(sys.stdin.readline())
arr = []

for i in range(n):
   arr.append([])
   m = int(input())
   for _ in range(m):
      arr[-1].append(list(map(int, sys.stdin.readline().rstrip().split())))


for i in solution(arr):
   print(i)







print(arr)

File: 백준/그리디/2/1946.js
const solution = (arrSize, data) => {
  let res = [];
  for (let i = 0; i < arrSize.length; i += 1) {
    let nowArr;
    if (i === 0) {
      nowArr = data.slice(0, arrSize[i]);
    } else {
      nowArr = data.slice(arrSize[i - 1], arrSize[i - 1] + arrSize[i]);
    }
    nowArr = nowArr.sort((a, b) => a[0] - b[0]);
    let std = nowArr[0][1];
    let tmp = 1;
    for (const el of nowArr) {
      if (el[1] < std) {
        tmp += 1;
        std = el[1];
      }
    }
    res.push(tmp);
  }
  for (const a of res) {
    console.log(a);
  }
};

const readline = require("readline");
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
const input = [];
rl.on("line", function (line) {
  input.push(line);
}).on("close", () => {
  let [_, ...arr] = input;
  const arrSize = [];
  const data = [];
  for (let i = 0; i < arr.length; i++) {
    if (arr[i].length < 2) {
      arrSize.push(+arr[i]);
    } else {
      const tmp = arr[i].split(" ").map((el) => +el);
      data.push(tmp);
    }
  }
  solution(arrSize, data);
});
